opal/runtime/opal_progress.c

```diff
@@ -229,6 +236,15 @@ opal_progress(void)
         }
     }
 
+    if (opal_progress_poll_when_idle) {
```

Comment:
Might want to make this an `OPAL_UNLIKELY`, since a) it won't be the common case, and b) it's in an extremely performance-sensitive part of the code.